### PR TITLE
Fix rule for linear-gradient with vendor prefix multivalue #2481

### PR DESCRIPTION
**Pull request by pfructuoso**
> Which issue, if any, is this issue related to?

#2481 

> Is there anything in the PR that needs further explanation?

It just splits the multiple properties to check them individualy

### CODE REVIEW

File: lib/rules/function-linear-gradient-no-nonstandard-direction/__tests__/index.js [hunk 1]
```diff
@@ -50,6 +50,33 @@ testRule(rule, {
     code: ".foo { background: -moz-linear-gradient(top, #fff, #000); }",
   }, {
     code: ".foo { background: -o-linear-gradient(top, #fff, #000); }",
+  }, {
+    code: ".foo { background: url(foo.png), -webkit-linear-gradient(bottom, #fff, #000 ); }",
+    message: messages.rejected,
```
Comment (jeddy3) [inline lib/rules/function-linear-gradient-no-nonstandard-direction/__tests__/index.js]: `accept` tests shouldn't take a `message` propertry

File: lib/rules/function-linear-gradient-no-nonstandard-direction/__tests__/index.js [hunk 2]
```diff
@@ -97,5 +124,32 @@ testRule(rule, {
   }, {
     code: ".foo { background: -o-linear-gradient(to bottom, #fff, #000 ); }",
     message: messages.rejected,
+  }, {
+    code: ".foo { background: url(foo.png), -webkit-linear-gradient(to bottom, #fff, #000 ); }",
+    message: messages.rejected,
```
Comment (jeddy3) [inline lib/rules/function-linear-gradient-no-nonstandard-direction/__tests__/index.js]: Please add `line` and `column` properties to all the `reject` tests - [see here](https://github.com/stylelint/stylelint/blob/master/lib/rules/at-rule-blacklist/__tests__/index.js#L25-L27) as an example.

File: lib/rules/function-linear-gradient-no-nonstandard-direction/index.js [hunk 3]
```diff
@@ -37,40 +38,45 @@ const rule = function (actual) {
...
-            index: expressionIndex,
-            result,
-            ruleName,
-          })
-        }
+          function complain() {
+            report({
+              message: messages.rejected,
+              node: decl,
+              index: expressionIndex,
```
Comment (jeddy3) [inline lib/rules/function-linear-gradient-no-nonstandard-direction/index.js]: I suspect you'll need to calculate the index based on the `expressionIndex` _and the index of the `postcss-value-parser` node_.

**Comment by pfructuoso** [comment]
@jeddy3 What do you mean by merging it to `v8` branch? Creating another PR for ´v8´ branch? I see it as a breaking change. The index problem is solved and it´s the same it was before. Nevertheless, if you need it or whatever other reason, just ask me again to do it and you´ll have it the faster possible!

File: lib/rules/function-linear-gradient-no-nonstandard-direction/index.js [hunk 4]
```diff
@@ -37,40 +38,45 @@ const rule = function (actual) {
...
-            index: expressionIndex,
-            result,
-            ruleName,
-          })
-        }
+          function complain() {
+            report({
+              message: messages.rejected,
+              node: decl,
+              index: decl.toString().indexOf(valueNode.value) + expressionIndex,
```
Comment (jeddy3) [inline lib/rules/function-linear-gradient-no-nonstandard-direction/index.js]: I think you'll want to make use of the [`declarationValueIndex` util](https://github.com/stylelint/stylelint/blob/master/lib/utils/declarationValueIndex.js) and the  [`node`'s `sourceIndex`](https://github.com/TrySound/postcss-value-parser#nodes) for consistency with the other rules, [like so](https://github.com/stylelint/stylelint/blob/master/lib/rules/declaration-property-unit-blacklist/index.js#L66), e.g.:

```js
index: declarationValueIndex(decl) + valueNode.sourceIndex + expressionIndex
```

**Comment by pfructuoso** [comment]
> The column and line numbers look good. I've made one further request to make the calculation of the index more consistent with other rules.

I like declarationValueIndex(decl)! I was looking for something like it but didn´t find it.

> I don't think you need to create another PR as you can use github to change the base branch. The v8 branch is for breaking changes. So, if possible, please set v8 as the base branch for this PR.

This is my first base branch change and now we have conflics. I don´t feel confortable trying to solve this in other branch without supervision. Could you guide me with this?

**Comment by jeddy3** [comment]
@pfructuoso Thanks for making the changes. They look good to me.

> I like declarationValueIndex(decl)! I was looking for something like it but didn´t find it.

Good stuff!

> This is my first base branch change and now we have conflics. I don´t feel confortable trying to solve this in other branch without supervision. Could you guide me with this?

@stylelint/core Can someone with better git-foo than I advise on how best to resolve this? Delete this branch and then just port @pfructuoso changes into a new one, or something else?

**Comment by pfructuoso** [comment]
> @stylelint/core Can someone with better git-foo than I advise on how best to resolve this? Delete this branch and then just port @pfructuoso changes into a new one, or something else?

I don´t know how to solve it without creating a new PR. I created a new [branch](https://github.com/pfructuoso/stylelint/tree/multivalue-linear-gradient-v8) from v8 and apply my changes but I can´t change the origin branch.

**Comment by hudochenkov** [comment]
@pfructuoso try to rename your old branch to something else, then rename new branch to `multivalue-linear-gradient` and force push.

**Comment by pfructuoso** [comment]
Ok, I did. The [branch](https://github.com/pfructuoso/stylelint/tree/multivalue-linear-gradient) exists but previous messages say it´ve been deleted and the PR closed. The reopen button is disabled. Is it only me or this doesn´t look good? Is it any problem on creating a new PR and referal it to this (#2482)?
screenshot

**Comment by hudochenkov** [comment]
@pfructuoso I'm so sorry. I believe I had same issues and did what I told you you to do. Maybe Github changed something and now behaviour is different.

**Comment by hudochenkov** [comment]
I've checked it's possible to open a new PR with base `v8`. But let's wait what others can say. I don't want to let you down again :(

**Pull request closed without merging**